src/components/DlTable/hooks/tablePagination.ts

```diff
@@ -54,16 +54,17 @@ export function useTablePaginationState(
     getCellValue: Function
 ) {
     const { props, emit } = vm
+
     const innerPagination = ref(
         Object.assign(
             {
                 sortBy: null,
                 descending: false,
                 page: 1,
                 rowsPerPage:
-                    props.rowsPerPageOptions.length > 0
-                        ? props.rowsPerPageOptions[0]
-                        : 5,
+                    props.pagination?.rowsPerPage ||
```

Comment:
If a user here specifies rowsPerpage to be 0 it will not take that value instead it will go and check other options.
the `||` should be `??` in order to allow people passing 0